README.md

```diff
@@ -1,3 +1,291 @@
+
+### Custom Class Prefixing
+
+You may have noticed the `FIS` prefix on the `FISAppDelegate` class in previous labs; these are the initials for "**F**lat**i**ron **S**chool", and is the customized prefix that we use in all of our Objective-C labs. It's best practice to prefix custom classes in order to disambiguate them from the classes in other libraries. While it's unlikely that you'd create a custom class that coincides with a class in an Apple framework, name collisions become a very real possibility when importing third party frameworks found in the Open Source community (i.e. via CocoaPods like the Specta and Expecta testing frameworks).
+
+Imagine that you're building a messaging app. Without class prefixing, you might create your own custom `Message` class. Now, if you decide to import another library that didn't use a class prefix either and which also includes a `Message` class, your application won't be able to distinguish between the two causing a slew of problems. However, if both you and the third party developer whose library you imported uniquely prefixed both `Message` classes (e.g. `FISMessage` and `JBMessage`), then the filenames would be unique. Furthermore, you might find yourself working with two libraries which both include a `Message` class, so it's similarly important that the classes are distinct:
+
+[JBMessage.h](https://github.com/josipbernat/JBMessage/blob/master/JBMessage/JBMessage/JBMessage.h)  
+[TSMessage.h](https://github.com/KrauseFx/TSMessages/blob/master/Pod/Classes/TSMessage.h)
+
+Prefixes are important in Objective-C so build the habit of using them early. Because of the testing suites, you'll have to keep to the `FIS` prefix when working on labs, but you should decide upon your own personal three-letter prefix for yourself when working on your own side projects, and a project- or group-specific prefix when working with a team.
+
+**Note:** *Apple reserves the authority to use any two-letter prefix for new libraries at future dates. For this reason, they suggest that third-party developers choose three-letter prefixes because there is no chance that Apple will create a future library with that exact prefix. This guideline is not always followed in the Open Source community, but it is considered best practice to use three-letter prefixes.*
+
+## A Time To Subclass
+
+Despite the plethora of class files available to us, solving any specific problem typically requires creating at least a few custom classes to tailor-fit our code to the current situation. None of the Apple libraries include, for example, a `Warship` class, nor a `ToySoldier` class, nor a `ChessBoard` class. In order to build an application that requires a representation of one of these objects, we would have to create a custom class to hold the data (properties) and perform the behaviors (methods) that the representation requires.
```

Comment:
In the end, almost all apps live within some sort of domain that is custom to them. A domain is the general area that your app lives in. Is it a shopping cart app? That means you'll need to model things like a `cart` and an `item`. no programming language could ever come with all classes for all possible problems! It is your job as a developer to model that domain, to bring all of the custom bits of the domain into your project